File: sdks/ts/src/sdk/models/shared/signrequestcreatesigner.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum SignRequestCreateSignerRoleEnum {
    Signer = "signer",
    Approver = "approver",
    FinalCopyReader = "final_copy_reader"
}


// SignRequestCreateSigner
/** 
 * The schema for a Signer in a POST Sign Request request body
**/
export class SignRequestCreateSigner extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=declined_redirect_url" })
  declinedRedirectUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=email" })
  email: string;

  @SpeakeasyMetadata({ data: "json, name=embed_url_external_user_id" })
  embedUrlExternalUserId?: string;

  @SpeakeasyMetadata({ data: "json, name=is_in_person" })
  isInPerson?: boolean;

  @SpeakeasyMetadata({ data: "json, name=login_required" })
  loginRequired?: boolean;

  @SpeakeasyMetadata({ data: "json, name=order" })
  order?: number;

  @SpeakeasyMetadata({ data: "json, name=password" })
  password?: string;

  @SpeakeasyMetadata({ data: "json, name=redirect_url" })
  redirectUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=role" })
  role?: SignRequestCreateSignerRoleEnum;

  @SpeakeasyMetadata({ data: "json, name=verification_phone_number" })
  verificationPhoneNumber?: string;
}

File: go-sdk/pkg/models/shared/signrequestcreatesigner.go
package shared

type SignRequestCreateSignerRoleEnum string

const (
	SignRequestCreateSignerRoleEnumSigner          SignRequestCreateSignerRoleEnum = "signer"
	SignRequestCreateSignerRoleEnumApprover        SignRequestCreateSignerRoleEnum = "approver"
	SignRequestCreateSignerRoleEnumFinalCopyReader SignRequestCreateSignerRoleEnum = "final_copy_reader"
)

// SignRequestCreateSigner
// The schema for a Signer in a POST Sign Request request body
type SignRequestCreateSigner struct {
	DeclinedRedirectURL     *string                          `json:"declined_redirect_url,omitempty"`
	Email                   string                           `json:"email"`
	EmbedURLExternalUserID  *string                          `json:"embed_url_external_user_id,omitempty"`
	IsInPerson              *bool                            `json:"is_in_person,omitempty"`
	LoginRequired           *bool                            `json:"login_required,omitempty"`
	Order                   *int64                           `json:"order,omitempty"`
	Password                *string                          `json:"password,omitempty"`
	RedirectURL             *string                          `json:"redirect_url,omitempty"`
	Role                    *SignRequestCreateSignerRoleEnum `json:"role,omitempty"`
	VerificationPhoneNumber *string                          `json:"verification_phone_number,omitempty"`
}
